### PR TITLE
SW-2608 Add IDs to buttons for analytics tracking

### PR DESCRIPTION
**Pull request by alexTerraformation**
Plan is to track some cancel button usage, but I added IDs to save/submit/download buttons as well in case we want them for the future.

### CODE REVIEW

File: src/components/accession2/edit/ViabilityModal.tsx [hunk 1]
```diff
@@ -71,8 +71,15 @@ export default function ViabilityDialog(props: ViabilityDialogProps): JSX.Elemen
         title={props.title}
         size='small'
         middleButtons={[
-          <Button label={strings.CANCEL} type='passive' onClick={onCloseHandler} priority='secondary' key='button-1' />,
-          <Button onClick={saveQuantity} label={strings.SAVE} key='button-2' />,
+          <Button
+            id='cancelViability'
```
Comment (karthikbtf) [inline src/components/accession2/edit/ViabilityModal.tsx:75]: suggestion: `cancelEditViabilityTest`

File: src/components/accession2/edit/ViabilityModal.tsx [hunk 2]
```diff
@@ -71,8 +71,15 @@ export default function ViabilityDialog(props: ViabilityDialogProps): JSX.Elemen
...
+            label={strings.CANCEL}
+            type='passive'
+            onClick={onCloseHandler}
+            priority='secondary'
+            key='button-1'
+          />,
+          <Button id='saveViability' onClick={saveQuantity} label={strings.SAVE} key='button-2' />,
```
Comment (karthikbtf) [inline src/components/accession2/edit/ViabilityModal.tsx:82]: suggestion: `saveEditViabilityTest`

File: src/components/accession2/viabilityTesting/DeleteViabilityTestModal.tsx [hunk 3]
```diff
@@ -36,8 +36,21 @@ export default function DeleteViabilityTestModal(props: DeleteViabilityTestModal
       title={strings.DELETE_VIABILITY_TEST}
       size='medium'
       middleButtons={[
-        <Button label={strings.CANCEL} type='passive' onClick={onCancel} priority='secondary' key='button-1' />,
-        <Button onClick={deleteHandler} type='destructive' label={strings.DELETE} key='button-2' />,
+        <Button
+          id='cancelDeleteViability'
```
Comment (karthikbtf) [inline src/components/accession2/viabilityTesting/DeleteViabilityTestModal.tsx:40]: suggestion: `cancelDeleteViabilityTest`

File: src/components/accession2/viabilityTesting/DeleteViabilityTestModal.tsx [hunk 4]
```diff
@@ -36,8 +36,21 @@ export default function DeleteViabilityTestModal(props: DeleteViabilityTestModal
...
+          label={strings.CANCEL}
+          type='passive'
+          onClick={onCancel}
+          priority='secondary'
+          key='button-1'
+        />,
+        <Button
+          id='deleteViability'
```
Comment (karthikbtf) [inline src/components/accession2/viabilityTesting/DeleteViabilityTestModal.tsx:48]: suggestion: `deleteViabilityTest`

File: src/components/seeds/database/DownloadReportModal.tsx [hunk 5]
```diff
@@ -59,7 +59,7 @@ export default function DownloadReportModal(props: DownloadReportModalProps): JS
           type='passive'
           onClick={handleCancel}
           key='button-1'
-          id='cancel'
+          id='cancelDownloadReport'
```
Comment (karthikbtf) [inline src/components/seeds/database/DownloadReportModal.tsx:62]: suggestion: `cancelDownloadAccessionsReport`

File: src/components/common/FormBottomBar.tsx [hunk 6]
```diff
@@ -60,6 +60,7 @@ export default function FormBottomBar({
       className={classes.bottomBar}
     >
       <Button
+        id='cancelBottomBar'
```
Comment (karthikbtf) [inline src/components/common/FormBottomBar.tsx:63]: This component is used in all those full page forms that have a save/cancel at the bottom of the page. 
We could maybe add more context to the component - in the form of a prop, just for the id purposes.
Current change is totally fine for now.

**Review by karthikbtf — COMMENTED** [review]
Looks good, some suggestions/comments (but totally up to you).
A few tests may need fixing (guessing the the test referred an id and cannot find it .. should be straightforward fixes).

File: src/components/common/FormBottomBar.tsx (hunk 6 above)
Comment (alexTerraformation) [reply src/components/common/FormBottomBar.tsx:63]: I had done what you suggested originally, but changed it back before committing.  You convinced me though - I will change it back.

**Review by karthikbtf — APPROVED** [review]
🙌  looks great! 
get some stats!!

**Pull request merged**